Src/FluentAssertions/Execution/AssertionScope.cs

```diff
@@ -44,9 +54,15 @@ private AssertionScope(IAssertionStrategy _assertionStrategy)
         public AssertionScope()
             : this(new CollectingAssertionStrategy())
         {
-            parent = current;
-            current = this;
 
+#if NOTNET45
```

Comment:
How about if we kept the use of `CallContext` and `AsyncLocal` a single place.
E.g.
```c#
private static AssertionScope CurrentImpl // I haven't thought about a good name
{
    get
    {
#if NET45
        return (AssertionScope) CallContext.LogicalGetData("this");
#else
        return current.Value;
#endif
    }
    set
    {
#if NET45
        CallContext.LogicalSetData("this", value);
#else
        current.Value = value;
#endif
    }
}
```

Reply:
done